File: pdu/python/geometry_msgs/pdu_cdr_conv_Point.py
from .pdu_pytype_Point import Point
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Point(writer: CdrWriter, src: Point):
    writer.write_float64(src.x)
    writer.write_float64(src.y)
    writer.write_float64(src.z)


def cdr_body_to_py_Point(reader: CdrReader, dst: Point):
    dst.x = reader.read_float64()
    dst.y = reader.read_float64()
    dst.z = reader.read_float64()
    return dst


def py_to_cdr_Point(src: Point) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Point(writer, src)
    return writer.bytes()


def cdr_to_py_Point(cdr_payload) -> Point:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Point()
    return cdr_body_to_py_Point(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Point.js
import { Point } from './pdu_jstype_Point.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduPointConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Point } src
     */
    static to_cdr_body(writer, src) {
        writer.write_float64(src.x);
        writer.write_float64(src.y);
        writer.write_float64(src.z);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Point } dst
     * @returns { Point }
     */
    static cdr_body_to_js(reader, dst) {
        dst.x = reader.read_float64();
        dst.y = reader.read_float64();
        dst.z = reader.read_float64();
        return dst;
    }

    /**
     * @param { Point } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Point }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Point());
    }
}
